src/clr-angular/data/datagrid/datagrid-row-detail.ts

```diff
@@ -67,14 +86,36 @@ export class ClrDatagridRowDetail<T = any> implements AfterContentInit, OnDestro
   public replacedRow = false;
 
   ngAfterContentInit() {
+    this.viewManager.detachAllViews(this._detailCells);
+    this.viewManager.insertAllViews(this._detailCells, this.assignRawOrders(), true);
     this.subscriptions.push(
       this.expand.replace.subscribe(replaceChange => {
         this.replacedRow = replaceChange;
+      }),
+      this.cells.changes.subscribe(() => {
+        this.viewManager.detachAllViews(this._detailCells);
+        this.viewManager.insertAllViews(this._detailCells, this.assignRawOrders(), true);
+      }),
+      // A subscription that listens for view reordering
+      this.columnReorderService.reorderRequested.subscribe(reorderRequest => {
```

Comment:
Suggestion, we could break up the subscriptions into smaller methods with descriptive names to help code readability. Something roughly like:

```typescript
this.subscriptions.push(
  this.replaceRowOnReplaceChanges(),
  this.updateViewsOnCellChange(),
  this.updateColumnOrderOnReorderChange()
);
```

Reply:
I like that. I will work on that.